sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1774,6 +1775,51 @@ class Analyzer(
   }
 
   /**
+   * Substitute Hints.
+   * - BROADCAST/BROADCASTJOIN/MAPJOIN match the closest table with the given name parameters.
+   */
+  object SubstituteHints extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case logical: LogicalPlan => logical transformDown {
+        case h @ Hint(name, parameters, child)
+            if Seq("BROADCAST", "BROADCASTJOIN", "MAPJOIN").contains(name.toUpperCase) =>
+          var resolvedChild = child
+
+          for (param <- parameters) {
+            val names = param.split("\\.")
```

Comment:
Sorry, I do not think this is beyond of the scope. Instead, this is a bug. This identifier stores a table identifier. If we do not use the same parsing solution, the result will be wrong.